subtle_training_bugs.py

```
@@ -0,0 +1,787 @@
+    if is_training:
+        model.train()
+    else:
+        model.eval()
+
+
+def train_with_mode_switching(model, train_data, val_data, train_target, val_target, optimizer):
+    """Training with mode switching"""
+    configure_model_mode(model, is_training=False)
+
+    val_output = model(val_data)
+    val_loss = nn.functional.cross_entropy(val_output, val_target)
```

Ruff (0.13.1)

Local variable `val_loss` is assigned to but never used

Remove assignment to unused variable `val_loss`

(F841)